src/test/java/org/opensearch/ad/rest/SecureADRestIT.java

```diff
@@ -65,14 +65,18 @@ public class SecureADRestIT extends AnomalyDetectorRestTestCase {
      * Create an unguessable password. Simple password are weak due to https://tinyurl.com/383em9zk
      * @return a random password.
      */
-    public static String generatePassword() {
-        String characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    public static String generatePassword(String username) {
+        String characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_";
 
         Random rng = new Random();
 
-        char[] password = new char[10];
-        for (int i = 0; i < 10; i++) {
-            password[i] = characters.charAt(rng.nextInt(characters.length()));
+        char[] password = new char[15];
+        for (int i = 0; i < 15; i++) {
+            char nextChar = characters.charAt(rng.nextInt(characters.length()));
+            while (username.indexOf(nextChar) > -1) {
```

Comment:
@kaituo I pushed a small change to update this conditional. In a previous commit it was `> 0` and I updated it to be `> -1`. The purpose for this loop is to ensure the randomly generated password does not share any characters in common with the username.

While testing I was getting WeakPasswordError on short usernames. If the username is `dog` and the randomly generated password had all the characters of dog it would complain they were too similar.